### DISCUSSION TITLE
Integrate with OpenHasp

### DISCUSSION THREAD

**Discussion by broyuken** [sle118/squeezelite-esp32#302] [Ideas]
Is there any possibility of combining the audio capabilities of squeezelite-32 with the touchscreen capabilities of OpenHasp? My end goal is to take a board like the W32-SC01 Plus and build a touch screen music box that can play music from whatever source (Spotify, LMS, etc) and see and control playback from the display. Hopefully we could eventually tie in with the Home Assistant Assist as well to have a fully local Alexa or google home alternative.

**Comment by philippe44** [comment]
LMS probably has that JiveLite that AFAIK can do what you want. But JiveLite would not run on just an esp32. It works on a Linux/FreeBSD/Windows and porting it to esp32 is a massive effort which includes adapting the graphic layer because the one it relies on does not exist on esp32. I had a brief look at it recently and decided that it's just not worth it.

The reason why the UI even on colored display of SqueezeESP32 is still very basic is that all the UI is driven by LMS. Anything else is either JIveLite or rebuild you own.

*I've edited a fair bit my initial answer that was clear as mud*

**Comment by broyuken** [comment]
But OpenHasp has all the needed display functionality built into it and runs on an esp32, it just can’t do the audio portion. Which is why I suggested combining both of these for an even better solution that covers both. Would it be possible to change what drives the display to whatever it is that openhasp uses?

The only reason I would think this wouldn’t be doable would be resources on the esp.

**Reply by philippe44** [reply]
Yea but still you need to write the UI and connect it to the "audio" layer. Think about just the music browser

**Reply by broyuken** [reply]
Isn’t the UI already written in the openhasp project?

**Reply by philippe44** [reply]
AFAIK, there is no full media library & control application. And you'd still need the bindings to LMS
